deepdive1/README.md

```diff
@@ -0,0 +1,265 @@
+
+## Testing Frameworks
+There are **tons** of testing frameworks. 
+* JUnit - Pretty much all the other testing frameworks are built off of this. JUnit is a testing library for Java in general that's widely used.
+* [AndroidJUnitRunner](https://developer.android.com/topic/libraries/testing-support-library/index.html#AndroidJUnitRunner) - JUnit compatible test runner for Android. Made by Google.
+* [Espresso](https://developer.android.com/topic/libraries/testing-support-library/index.html#Espresso) - UI testing framework. Good for functional UI testing within an app. Only supports API level 8, 10, => 15. Made by Google.
+* [UI Automator](https://developer.android.com/topic/libraries/testing-support-library/index.html#UIAutomator) - UI testing framework. Good for cross-app functional UI testing. Only supports API levels => 16. Made by Google.
+* [Robolectric](http://robolectric.org/) - Lets you run Android tests directly from inside your IDE instead of in an emulator
+* [Robotium](https://github.com/robotiumtech/robotium) - Similar to [Selenium](https://en.wikipedia.org/wiki/Selenium_(software)) (a testing framework for web), it lets you make robust automatic black-box UI tests.
+* [Calabash](https://github.com/calabash/calabash-android) - It has an English-like syntax that makes it very easy to understand. The tests are described in [Cucumber](https://cucumber.io/) and then converted to Robotium in run time.
+* [Appium](http://appium.io/) - You can test both iOS and Android versions of your app at the same time.
+* And probably many more...
+
+## Making your first local test
+*Note: This is a condensed version of [this](https://developer.android.com/training/testing/unit-testing/local-unit-tests.html#build).*
+
+Let's say I have a certain class I want to test. For this example, I'll use `MathExamples.java` as my class to test:
+
+```
+public class MathExamples {
+
+    public int fibonacci(int n){
+        if (n==1 || n==2) return 2;
+        else return fibonacci(n-1) + fibonacci(n-2);
+    }
+
+    public int addition(int a, int b){
+        return a + b;
+    }
+
+    public int multiplyByTen(int a){
+        return 10 * a;
+    }
+}
+```
+
+There's a little mistake in there ;-)
+
+### 1. Check build.gradle for dependencies
+Depending on which framework you use, the steps here will be different. For now, let's include these dependencies in our build.gradle:
+
+```
+dependencies {
+    // Required -- JUnit 4 framework
+    testCompile 'junit:junit:4.12'
+    // Optional -- Mockito framework
+    testCompile 'org.mockito:mockito-core:1.10.19'
+    // Optional -- Hamcrest library
+    testCompile 'org.hamcrest:hamcrest-library:1.3'
+}
+```
+
+[Mockito](http://mockito.org/) is a framework that lets you create mock variables for things that don't always stay the same. For example, your app's `context` might be something you want to mock.
```

Comment:
I wanted to include a little bit more about how to use it, so I was thinking of maybe writing an extra bit about how to use it